framework/CodeInterfaces/RAVEN/RAVENInterface.py

```diff
@@ -0,0 +1,257 @@
+      This method should be overwritten if these are not acceptable defaults.
+      @ In, None
+      @ Out, None
+    """
+    self.addInputExtension(['xml'])
+
+  def _readMoreXML(self,xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class and initialize
+      some members based on inputs. This can be overloaded in specialize code interface in order to
+      read specific flags.
+      In this case, this is used for locating an external python module where the variables can be modified and the "vector" variables
+      can be splitted in multiple single variables
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None.
+    """
+    if os.path.basename(xmlNode.find("executable").text) != 'raven_framework':
+      raise IOError(self.printTag+' ERROR: executable must be "raven_framework" (in whatever location)!')
+
+    linkedDataObjects = xmlNode.find("outputExportOutStreams")
+    if linkedDataObjects is None:
+      raise IOError(self.printTag+' ERROR: outputExportOutStreams node not present. You must input at least one OutStream (max 2)!')
+    self.linkedDataObjectOutStreamsNames = linkedDataObjects.text.split(",")
+    if len(self.linkedDataObjectOutStreamsNames) > 2:
+      raise IOError(self.printTag+' ERROR: outputExportOutStreams node. The maximum number of linked OutStreams are 2 (1 for PointSet and 1 for HistorySet)!')
+
+    child = xmlNode.find("conversionModule")
+    if child is not None:
+      extModForVarsManipulationPath = os.path.expanduser(child.text.strip())
+      if not os.path.isabs(extModForVarsManipulationPath):
+        extModForVarsManipulationPath = os.path.abspath(extModForVarsManipulationPath)
+      # check if it exist
+      if not os.path.exists(extModForVarsManipulationPath):
+        raise IOError(self.printTag+' ERROR: the conversionModule "'+extModForVarsManipulationPath+'" has not been found!')
+      self.extModForVarsManipulation = utils.importFromPath(extModForVarsManipulationPath)
+      if self.extModForVarsManipulation is None:
+        raise IOError(self.printTag+' ERROR: the conversionModule "'+extModForVarsManipulationPath+'" failed to be imported!')
+      # check if the methods are there
+      if 'convertNotScalarSampledVariables' in self.extModForVarsManipulation.__dict__.keys():
+        self.hasMethods['noscalar'] = True
+      if 'manipulateScalarSampledVariables' in self.extModForVarsManipulation.__dict__.keys():
+        self.hasMethods['scalar'  ] = True
+      if not self.hasMethods['scalar'] and not self.hasMethods['noscalar']:
+        raise IOError(self.printTag +' ERROR: the conversionModule "'+extModForVarsManipulationPath
+                                    +'" does not contain any of the usable methods! Expected at least '
+                                    +'one of: "manipulateScalarSampledVariables" and/or "manipulateScalarSampledVariables"!')
```

Comment:
I would like to keep this as an error since the only reason why a user should "import" that module is to implement one of those 2 methods. Otherwise the interface is not going to do anything with it.